### PR TITLE
1879 edit operation

### PR DESCRIPTION
**Pull request by marcellmueller**
#1879 
To test this make sure to reset the database since it includes updated fixtures:

- `cd bc_obps && make reset_db && make run`
- Navigate to the `Operations` dashboard in the `Administration` module
- Select any Operation that has `View Operation` in the action column and click `View Operation` (made [an issue](https://github.com/bcgov/cas-registration/issues/2270) to fix the `Start Registration` and `Continue Registration` URLs)
- Click `Edit` button at the bottom of the form
- Make any necessary changes (file upload fields will definitely be required)
- Click `Submit`
- The form should save. You can refresh the page and the updates will remain.

Other things to test:
- Click `View Details` for Operation 14 which has a purpose of `Electricity Import Operation`
- The `Regulated product names` field does not appear under `Registration Purpose`
- Return to Operations DataGrid, click `View Details` for Operation 14 which has a purpose of `Potential Reporting Operation`
- The `Regulated product names` field does not appear under `Registration Purpose`
- The Multiple Operators section works

### CODE REVIEW

File: bc_obps/registration/api/v2/_operations/operation_id.py [hunk 1]
```diff
@@ -29,3 +30,33 @@
...
+    "/v2/operations/{uuid:operation_id}/with-documents",
+    response={200: OperationOutWithDocuments, codes_4xx: Message},
+    tags=OPERATION_TAGS,
+    description="""Retrieves the details of a specific operation by its ID along with it's documents""",
+    exclude_none=True,
+    auth=authorize("approved_authorized_roles"),
+)
+@handle_http_errors()
+def get_operation_with_documents(request: HttpRequest, operation_id: UUID) -> Tuple[Literal[200], Operation]:
+    return 200, OperationService.get_if_authorized(get_current_user_guid(request), operation_id)
```
Comment (marcellmueller) [inline bc_obps/registration/api/v2/_operations/operation_id.py:45]: Added a new API route, though I was split on if I should add a query param like `documents` and return the documents if it was true. Can we have conditional Django Ninja schemas? Is a separate route fine?

**Comment by shon-button** [comment]
@marcellmueller 

Re #2287 
Error is also seen on client.
image

image

**Comment by shon-button** [comment]
@marcellmueller 

Q: are the uploaded files supposed to have download links?

Thanks

**Comment by marcellmueller** [comment]
> @marcellmueller
> 
> Re #2287 FYI: Error is also seen on client. image
> 
> image

There is a bug with the Read only file widget, it's not unique to this issue:
https://github.com/bcgov/cas-registration/issues/2287

**Comment by marcellmueller** [comment]
> @marcellmueller
> 
> Q: are the uploaded files supposed to have download links?
> 
> Thanks

https://github.com/bcgov/cas-registration/pull/2253/commits/49782ddbb9e96f720ea442cd5f737a1c1167a7f9

Added the file preview prop! Thanks.

File: bc_obps/service/operation_service_v2.py
Comment (shon-button) [inline bc_obps/service/operation_service_v2.py:1]: @marcellmueller 

FYC, :)

Perhaps the backend service should verify that `regulated_products` should only be set for specific registration purposes (e.g., ELECTRICITY_IMPORT_OPERATION and POTENTIAL_REPORTING_OPERATION)

File: bc_obps/registration/tests/endpoints/v2/_operations/test_operation_id.py
Comment (shon-button) [inline bc_obps/registration/tests/endpoints/v2/_operations/test_operation_id.py:1]: @marcellmueller 

nit: perhaps ` def test_industry_users_can_only_get_their_own_operations_with_documents(self)` should be two separate tests:

`def test_industry_user_can_get_owned_operation_with_documents(self):`
`def test_industry_user_cannot_get_unowned_operation_with_documents(self):`

File: bc_obps/service/operation_service_v2.py
Comment (marcellmueller) [reply bc_obps/service/operation_service_v2.py:1]: I asked Brianna about this and she said not to worry about it seeing as all of the purpose conditionals are changing re: the business area.

File: bc_obps/registration/tests/endpoints/v2/_operations/test_operation_id.py
Comment (marcellmueller) [reply bc_obps/registration/tests/endpoints/v2/_operations/test_operation_id.py:1]: I don't spend much time in here and was just following the patterns set in the `get_operation` test above this. However I can update this if you think it's a good use of my time.

**Review by shon-button — APPROVED** [review]

File: bc_obps/service/operation_service_v2.py
Comment (marcellmueller) [reply bc_obps/service/operation_service_v2.py:1]: I added a comment here to remind us to potentially add that conditional in the future after the create registration purpose service gets redone. Thanks for the heads up, this was a good issue to raise 🙇

**Pull request merged**